free-2.md

```diff
@@ -11,41 +11,90 @@ _What need is met by your idea? Who are you helping? What is that person's
 experience like now? What would their experience be like if you could help
 them?_
 
+My idea is to do the "Re-design / re-implement an existing DSL" type of project
+of the Harvey Mudd Miniature Machine (HMMM), the assembly language used in CS5
+here at HMC.
+
+Through this language, students in CS5 who are learning about how a computer
+works are learning about the basics of assembly language by getting to use a
+simplified version of the language themselves. If they did not have a language
+to practice with, it would be much more difficult to learn how assembly works.
+
 ### Why a language?
 
 _Why is a DSL appropriate for your user(s)? How does it address the need?_
 
+Giving students in CS5 real assembly language would be very over-complicated,
+there is a lot to learn in CS5 and it is probably best to keep the content
+very focused and teach only aspects of assembly language that students must learn
+about. It addresses this need by giving students a simplified assembly language
+that they can run and debug.
+
 ### Why you?
 
 _What excites you about this idea? How did you come up with it?_
 
+I liked the idea of doing a re-implementation project described in the homework,
+and I thought that this might be a unique project to apply the re-implementation
+technique to. I came up with it because I was thinking about different smaller
+programs that I might want to re-implement, thought of HMMM, read the
+description page, and thought it would be a good fit!
+
 ### Domain
 
 _Describe the project's domain in five words._
 
+Assembly language for CS5 learning.
+
 ### Interface (syntax)
 
 _How might the user interact with the language? What does programming look
 like? Why is this the right way to interact with the problem domain?_
 
+The user would be able to code in the assembly language, then have it be
+assembled, and then run. There should also be a debugging interface, where the
+user can walk through each line, and the program will show what information is
+in the registers and the stack as each line is executed.
```

Comment:
If you're planning to do this w/ JS, I recommend using the Ace Editor!
https://ace.c9.io/